### PR TITLE
Add block editor enhancements, block styles and color scheme palettes

### PR DESCRIPTION
**Pull request by richtabor**
The PR introduces full editor styling within Gutenberg, additional styling for converting Beaver Building templates to Gutenberg blocks, and new block color palettes. 

Here's a gif of the new color palettes in action:
ScreenFlow

### CODE REVIEW

**Comment by richtabor** [comment]
Here's a gif of the Customizer styles being pulled into the editor as well: 
ScreenFlow

**Comment by EvanHerman** [comment]
@richtabor This looks good to me 👍 I'm just going through some manual testing now. Can we fix up these travis builds?

**Comment by EvanHerman** [comment]
@richtabor Investigating this a bit further, but logging this here. I was running TwentySeventeen and switched over to Primer and see the following in the customizer:

`Notice: Undefined index: default in /Users/eherman/Desktop/Sites/primer/wp-content/themes/primer/inc/customizer/colors.php on line 1238`

Example

The `$this->color_schemes` array is empty.

File: inc/compat/gutenberg.php [hunk 1]
```diff
@@ -16,24 +16,32 @@
...
-function primer_gutenberg_editor_assets() {
-
-	global $post;
+function primer_editor_frame_styles() {
 
 	$suffix = SCRIPT_DEBUG ? '' : '.min';
-
-	wp_enqueue_style( 'primer-gutenberg-editor', get_template_directory_uri() . "/assets/css/admin/gutenberg-editor{$suffix}.css", true, defined( 'PRIMER_CHILD_VERSION' ) ? PRIMER_CHILD_VERSION : PRIMER_VERSION, 'all' );
-
+	wp_enqueue_style( Primer_Customizer::$stylesheet . '-editor-frame', get_template_directory_uri() . "/assets/css/admin/editor-frame{$suffix}.css", array(), PRIMER_VERSION, 'all' );
```
Comment (EvanHerman) [inline inc/compat/gutenberg.php:47]: @richtabor Add a space here between `$suffix` and `wp_enqueue_style` and then again between `wp_enqueue_style` and the closing `}` of the function.

File: inc/compat/gutenberg.php [hunk 2]
```diff
@@ -16,24 +16,32 @@
...
 
+	// Enqueue editor styles if the block editor exists.
+	if ( function_exists( 'register_block_type' ) ) {
+
+		$suffix = SCRIPT_DEBUG ? '' : '.min';
+
+		add_theme_support( 'editor-styles' );
+
+		// Load regular editor styles into the new block-based editor.
+		add_editor_style( "assets/css/admin/editor-blocks{$suffix}.css" );
```
Comment (EvanHerman) [inline inc/compat/gutenberg.php:33]: Add a space here between `add_editor_style` and the closing `}`

File: functions.php [hunk 3]
```diff
@@ -374,12 +376,25 @@ function primer_content_width() {
...
  *
  * @link  https://developer.wordpress.org/reference/functions/add_editor_style/
  * @since 1.0.0
  */
-add_action( 'admin_init', 'add_editor_style', 10, 0 );
+function primer_classic_editor_styles() {
+
+	// Return if the block editor is not found.
+	if ( ! function_exists( 'register_block_type' ) ) {
+		return;
```
Comment (EvanHerman) [inline functions.php:389]: Double space this

```php
if ( ! function_exists( 'register_block_type' ) ) {

		return;

}

File: functions.php [hunk 4]
```diff
@@ -374,12 +376,25 @@ function primer_content_width() {
...
+	}
+
+	// Add editor styles for the classic editor.
+	if ( ! get_current_screen()->is_block_editor() ) {
+		add_editor_style( 'editor-style.css' );
```
Comment (EvanHerman) [inline functions.php:396]: Double space this as well.

File: inc/customizer/colors.php [hunk 5]
```diff
@@ -427,8 +478,77 @@ public function __construct() {
 						.site-info-wrapper .social-menu a:visited,
 						.site-info-wrapper .social-menu a:hover,
 						.site-info-wrapper .social-menu a:visited:hover' => array(
+	'color' => '%1$s',
+),
```
Comment (EvanHerman) [inline inc/customizer/colors.php]: The spacing is off on this line.

File: inc/customizer/colors.php [hunk 6]
```diff
@@ -276,63 +300,85 @@ public function __construct() {
...
 						'button:hover, button:active, button:focus,
 						a.button:hover, a.button:active, a.button:focus, a.button:visited:hover, a.button:visited:active, a.button:visited:focus,
 						input[type="button"]:hover, input[type="button"]:active, input[type="button"]:focus,
 						input[type="reset"]:hover, input[type="reset"]:active, input[type="reset"]:focus,
 						input[type="submit"]:hover, input[type="submit"]:active, input[type="submit"]:focus' => array(
-							'background-color' => 'rgba(%1$s, 0.8)',
-							'border-color'     => 'rgba(%1$s, 0.8)',
+	'background-color' => 'rgba(%1$s, 0.8)',
+	'border-color'     => 'rgba(%1$s, 0.8)',
+),
```
Comment (EvanHerman) [inline inc/customizer/colors.php]: Spacing is also off here.

File: inc/customizer/colors.php [hunk 7]
```diff
@@ -276,63 +300,85 @@ public function __construct() {
...
 						input[type="reset"],
 						input[type="submit"],
+						.wp-block-button__link,
 						.site-info-wrapper .social-menu a:hover' => array(
-							'background-color' => '%1$s',
-							'border-color'     => '%1$s',
-						),
+	'background-color' => '%1$s',
+	'border-color'     => '%1$s',
+),
```
Comment (EvanHerman) [inline inc/customizer/colors.php]: The spacing is off here

File: inc/customizer/colors.php [hunk 8]
```diff
@@ -276,63 +300,85 @@ public function __construct() {
...
+					'label'           => esc_html__( 'Link Text', 'primer' ),
+					'default'         => '#ff6663',
+					'section'         => 'colors-content',
+					'css'             => array(
 						'a, a:visited,
 						.entry-title a:hover, .entry-title a:visited:hover' => array(
-							'color' => '%1$s',
-						),
+	'color' => '%1$s',
+),
```
Comment (EvanHerman) [inline inc/customizer/colors.php]: Fix spacing on this line

File: inc/customizer/colors.php [hunk 9]
```diff
@@ -248,17 +272,17 @@ public function __construct() {
...
 							'background-color' => '%1$s',
 						),
 					),
 					'rgba_css' => array(
 						'.footer-menu ul li a:hover,
 						.footer-menu ul li a:visited:hover' => array(
-							'color' => 'rgba(%1$s, 0.8)',
-						),
+	'color' => 'rgba(%1$s, 0.8)',
+),
```
Comment (EvanHerman) [inline inc/customizer/colors.php]: Spacing

File: inc/customizer/colors.php [hunk 10]
```diff
@@ -248,17 +272,17 @@ public function __construct() {
 					'css'             => array(
 						'.footer-menu ul li a,
 						.footer-menu ul li a:visited' => array(
-							'color' => '%1$s',
-						),
+	'color' => '%1$s',
+),
```
Comment (EvanHerman) [inline inc/customizer/colors.php]: Spacing

File: inc/customizer/colors.php [hunk 11]
```diff
@@ -235,8 +259,8 @@ public function __construct() {
 					'css'             => array(
 						'.site-footer .widget,
 						.site-footer .widget form label' => array(
-							'color' => '%1$s',
-						),
+	'color' => '%1$s',
+),
```
Comment (EvanHerman) [inline inc/customizer/colors.php]: Spacing

File: inc/customizer/colors.php [hunk 12]
```diff
@@ -205,9 +222,16 @@ public function __construct() {
...
+						.logged-in-as,
+						.wp-block-coblocks-author__heading' => array(
+	'color' => '%1$s',
+),
+					),
+					'editor_css'         => array(
+						'.wp-block-quote,
+						.wp-block-coblocks-author__heading' => array(
+	'color' => '%1$s',
+),
```
Comment (EvanHerman) [inline inc/customizer/colors.php]: Spacing

File: inc/customizer/colors.php [hunk 13]
```diff
@@ -205,9 +222,16 @@ public function __construct() {
 						.entry-meta,
 						.entry-footer,
 						.comment-meta .says,
-						.logged-in-as' => array(
-							'color' => '%1$s',
-						),
+						.logged-in-as,
+						.wp-block-coblocks-author__heading' => array(
+	'color' => '%1$s',
+),
```
Comment (EvanHerman) [inline inc/customizer/colors.php]: Spacing

File: inc/customizer/colors.php [hunk 14]
```diff
@@ -175,10 +187,10 @@ public function __construct() {
 						input[type="color"]:focus,
 						textarea:focus,
 						.navigation.pagination .paging-nav-text' => array(
-							'color' => '%1$s',
-						),
+	'color' => '%1$s',
+),
```
Comment (EvanHerman) [inline inc/customizer/colors.php]: Spacing

File: inc/customizer/colors.php [hunk 15]
```diff
@@ -139,14 +140,25 @@ public function __construct() {
...
+					'editor_css'  => array(
+						'.editor-styles-wrapper .editor-post-title__block textarea.editor-post-title__input,
+						.editor-styles-wrapper .wp-block h1,
+						.editor-styles-wrapper .wp-block h2,
+						.editor-styles-wrapper .wp-block h3,
+						.editor-styles-wrapper .wp-block h4,
+						.editor-styles-wrapper .wp-block h5,
+						.editor-styles-wrapper .wp-block h6' => array(
+	'color' => '%1$s',
+),
```
Comment (EvanHerman) [inline inc/customizer/colors.php]: Spacing

File: inc/customizer/colors.php [hunk 16]
```diff
@@ -139,14 +140,25 @@ public function __construct() {
 						dl dt,
 						.entry-title, .entry-title a, .entry-title a:visited,
 						.widget-title' => array(
-							'color' => '%1$s',
-						),
+	'color' => '%1$s',
+),
```
Comment (EvanHerman) [inline inc/customizer/colors.php]: Spacing

File: inc/customizer/colors.php [hunk 17]
```diff
@@ -122,8 +123,8 @@ public function __construct() {
 					'rgba_css' => array(
 						'.main-navigation ul li a:hover,
 						.main-navigation ul li a:focus' => array(
-							'color' => 'rgba(%1$s, 0.8)',
-						),
+	'color' => 'rgba(%1$s, 0.8)',
+),
```
Comment (EvanHerman) [inline inc/customizer/colors.php]: Spacing

File: inc/customizer/colors.php [hunk 18]
```diff
@@ -470,58 +590,120 @@ public function __construct() {
...
+				'label'      => esc_html__( 'Default', 'primer' ),
+				'colors'     => array_combine(
 					array_keys( $this->colors ),
 					wp_list_pluck( $this->colors, 'default' )
 				),
+				'primary'    => sanitize_hex_color( '#' . get_theme_mod( 'primary_color', $this->get_default_color( 'primary_color', 'default' ) ) ),
+				'secondary'  => sanitize_hex_color( '#' . get_theme_mod( 'secondary_color', $this->get_default_color( 'secondary_color', 'default' ) ) ),
+				'tertiary'   => sanitize_hex_color( '#' . get_theme_mod( 'tertiary_color', $this->get_default_color( 'tertiary_color', 'default' ) ) ),
+				'quaternary' => sanitize_hex_color( '#' . get_theme_mod( 'quaternary_color', $this->get_default_color( 'quaternary_color', 'default' ) ) ),
+				'quinary'    => sanitize_hex_color( '#' . get_theme_mod( 'quinary_color', $this->get_default_color( 'quinary_color', 'default' ) ) ),
```
Comment (EvanHerman) [inline inc/customizer/colors.php]: It looks like this section is causing the errors that I mentioned in slack this morning. `$this->color_schemes` looks like it's empty in `$this->get_default_color` run in the construction of `colors.php`

**Review by EvanHerman — APPROVED** [review]

**Pull request merged**